### ISSUE TITLE
Cannot access CDN profiles when using `AppServiceMSICredentials`+`withDefaultSubscription`.

### ISSUE THREAD

**Issue by yevster** [Azure/azure-libraries-for-java#775]
**Describe the bug**

***Exception or Stack Trace***
Add the exception log and stack trace if available

**To Reproduce**

***Code Snippet***
Add the code snippet that causes the issue.

**Expected behavior**

**Screenshots**

**Setup (please complete the following information):**
 - IDE : [e.g. IntelliJ]
 - Version of the Library used

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by maggiepint** [comment]
Thanks for reporting this. I have routed to the appropriate team.

**Comment by yaohaizh** [comment]
@yevster AZure cli only set your local subscription while azure function cannot pick it. You might need set the subscripionid in your Azure function? Could you try it and see whether it can be succeeded?

**Comment by yevster** [comment]
@yaohaizh, What do you mean by "set the subscriptionid in your Azure function"? Do I need to change my original code snippet?

**Comment by yaohaizh** [comment]
```
AppServiceMSICredentials(AzureEnvironment.AZURE)).WithSubscription(subscriptionId)
```

**Comment by yevster** [comment]
Yes, that works - it was the workaround I used. However, this is still a bug.

First, the dependency on Azure CLI is not documented in [the documentation for `withDefaultSubscription`](https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.management.azure.authenticated.withdefaultsubscription?view=azure-java-stable).

Second, if `withDefaultSubscription` is used in an illegal context, an exception should be thrown. As is, the SDK is failing non-transparrently when performing a legal operation.

**Comment by yaohaizh** [comment]
@yevster  you can use AzureCliCredential now. Please let us know whether this fit your need.  Thanks.
